windows/installer_jack.nsi

```diff
@@ -0,0 +1,443 @@
+            Abort
+        ${EndIf}
+
+    continue_${UniqueID}:
+    !undef UniqueID
+
+    ; Install folders and files
+    CreateDirectory "$INSTDIR"
+    !include "${files}"
+
+    ; Add the redistribution license
+    File "/oname=$INSTDIR\COPYING" "${ROOT_PATH}\COPYING"
+    File "/oname=$INSTDIR\servericon.ico" "${SERVER_ICON}"
+
+    ; Cleanup
+    !delfile "${files}"
+    !undef files
+    !undef prefix
+
+!macroend
+
+!macro SetupShortcuts
+
+    ; Add the registry key to store the installation folder
+    WriteRegStr HKLM "${APP_INSTALL_KEY}" "${APP_INSTALL_VALUE}" "$INSTDIR"
+
+    ; Add the registry keys so that software appears in Windows "Add/Remove Software"
+    WriteRegStr HKLM "${APP_UNINSTALL_KEY}" "DisplayName"     "${APP_NAME}"
+    WriteRegStr HKLM "${APP_UNINSTALL_KEY}" "DisplayIcon"     "$INSTDIR\${APP_EXE},0"
+    WriteRegStr HKLM "${APP_UNINSTALL_KEY}" "UninstallString" '"$INSTDIR\${UNINSTALL_EXE}"'
+
+    ; Add the uninstaller
+    WriteUninstaller "$INSTDIR\${UNINSTALL_EXE}"
+
+    ; Add the Start Menu shortcuts
+    CreateDirectory "$SMPROGRAMS\${APP_NAME}"
+    CreateShortCut  "$SMPROGRAMS\${APP_NAME}\${APP_NAME}.lnk"           "$INSTDIR\${APP_EXE}"
+    CreateShortCut  "$SMPROGRAMS\${APP_NAME}\${APP_NAME} Server.lnk"    "$INSTDIR\${APP_EXE}" "-s" "$INSTDIR\servericon.ico"
+    CreateShortCut  "$SMPROGRAMS\${APP_NAME}\${APP_NAME} Uninstall.lnk" "$INSTDIR\${UNINSTALL_EXE}"
+
+!macroend
+
+!macro SecSelect SecId ; See https://nsis.sourceforge.io/Managing_Sections_on_Runtime
+    Push $0
+    IntOp $0 ${SF_SELECTED} | ${SF_RO}
+    SectionSetFlags ${SecId} $0
+    SectionSetInstTypes ${SecId} 1
+    Pop $0
+!macroend
+
+!define SelectSection '!insertmacro SecSelect'
+
+!macro SecUnSelect SecId
+  Push $0
+  IntOp $0 ${SF_USELECTED} | ${SF_RO}
+  SectionSetFlags ${SecId} $0
+  SectionSetText  ${SecId} ""
+  Pop $0
+!macroend
+
+!define UnSelectSection '!insertmacro SecUnSelect'
+
+Section "Install_64Bit" INST_64
+    ; check if old, wrongly installed Jamulus exists. See https://stackoverflow.com/questions/27839860/nsis-check-if-registry-key-value-exists#27841158
+    IfFileExists "$PROGRAMFILES32\Jamulus\Uninstall.exe" 0 continueinstall
+
+        MessageBox MB_YESNOCANCEL|MB_ICONEXCLAMATION "$(OLD_WRONG_VER_FOUND)" /sd IDYES IDNO idontcare IDCANCEL quit
+            goto removeold
+
+        idontcare: ; Clicked no
+            MessageBox MB_YESNO|MB_ICONEXCLAMATION "$(OLD_WRONG_VER_FOUND_CONFIRM)" /sd IDNO IDYES continueinstall
+            goto removeold
+
+        removeold: ; Remove it
+            ExecWait '"$PROGRAMFILES32\Jamulus\Uninstall.exe" /S' $0
+            ${IfNot} $0 == 0
+                MessageBox MB_OKCANCEL|MB_ICONEXCLAMATION "$(OLD_VER_REMOVE_FAILED)" /sd IDCANCEL IDOK continueinstall
+                goto quit
+            ${EndIf}
+
+            goto continueinstall
+
+        quit:
+            Abort
+
+    continueinstall:
```

Comment:
This could be removed in theory (since we could see this Jamulus Version as something totally different)

Reply:
I don't mind removing this, but could this still help in cases you currently have an old version with ASIO and you now decide to install the JACK version (because it has been made available now)?

Reply:
I think we should provide uninstalling any versions if possible.
Previous, ASIO, wrong install...

Reply:
Hmm. Probably true. But I'm in favor of removing it some day (it's just added logic we won't need in future). Yes, it does break backwards-compatibility, so maybe we'll need to wait a bit more.